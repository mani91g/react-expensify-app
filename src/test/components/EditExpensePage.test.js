import React from 'react'
import {shallow} from 'enzyme'
import {EditExpensePage} from '../../components/EditExpensePage'
import expenses from '../fixtures/expenses'

let wrapper, editExpense, removeExpense, history

beforeEach(() => {
    editExpense = jest.fn()
    removeExpense = jest.fn()
    history = {push: jest.fn()}
    wrapper = shallow(
        <EditExpensePage 
        editExpense={editExpense} 
        history={history} 
        removeExpense = {removeExpense}
        expense={expenses[1]}/>)
})

test('Should render edit test page', () => {
    expect(wrapper).toMatchSnapshot()
})

test('Should test on edit expense onSubmit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1])
    expect(editExpense).toHaveBeenLastCalledWith(expenses[1].id,expenses[1])
    expect(history.push).toHaveBeenLastCalledWith('/')
})

test('Should test on edit expense remove expense', () => {
    wrapper.find('button').simulate('click')
    expect(removeExpense).toHaveBeenLastCalledWith({id: expenses[1].id})
    expect(history.push).toHaveBeenLastCalledWith('/')
})