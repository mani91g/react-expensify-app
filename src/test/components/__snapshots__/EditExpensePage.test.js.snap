// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render edit test page 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 12000,
        "createdAt": 345600000,
        "description": "houserent",
        "id": "2",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
